index.js

```diff
@@ -368,6 +371,67 @@ module.exports = {
         // prettier-ignore
         if (group.excluded) return deferredSource.abort( new Error("We're excluded from this group, can't list members"))
 
+        if (allAdded) {
+          // TODO: make secrets live. box2 live get?
+          const secrets = group.readKeys.map((readKey) =>
+            readKey.key.toString('base64')
+          )
+
+          pull(
+            ssb.db.query(
+              where(
+                and(
+                  isDecrypted('box2'),
+                  type('group/add-member'),
+                  groupRecp(groupId)
+                )
+              ),
+              dbLive({ old: true }),
+              toPullStream()
+            ),
+            pull.filter(isAddMember),
+            pull.take(1),
+            pull.collect((err, [addMsg]) => {
+              // prettier-ignore
+              if (err) return deferredSource.abort(clarify(err, "Error getting add-member message for group we're listing all members of"))
+              // prettier-ignore
+              if (!addMsg) return deferredSource.abort(new Error("Missing add-member message for group we're listing all members of"))
+
+              const creator = addMsg.value.content.creator
+
+              const source = pull(
+                ssb.db.query(
+                  where(
+                    and(
+                      type('group/init'),
+                      slowPredicate(
+                        'secret',
+                        (secret) => secrets.includes(secret),
+                        { name: secrets.join('') }
+                      )
+                    )
+                  ),
+                  live ? dbLive({ old: true }) : null,
+                  toPullStream()
+                ),
+                pull.filter((msg) => isInitRoot(msg) || isInitEpoch(msg)),
+                pull.map((msg) => msg.key),
+                pull.unique(),
+                pull.map((epochRootId) =>
+                  getMembers.stream(epochRootId, { live })
```

Comment:
we need to adapt this function to getMembers having this `{added, toExclude}` format